### DISCUSSION TITLE
which file is downloaded using web ui ?

### DISCUSSION THREAD

**Discussion by yarez0** [immich-app/immich#10497] [General]
hi guys, 

i'm testing this great app before migrating from piwigo to it but i'm wondering which file is downloaded when using the web ui button ?

my test is to backup an album from my ios device to an album, using the storage template migration 
the naming convention is correct : {{{album}}}/{{y}}{{MM}}{{dd}}_{{HH}}{{mm}}{{SSS}}
the "i" button is showing the good file path and file name

but if i want to download the photo, the photo i download is named by its backup name "IMG_xxxx"

<img width="508" alt="Capture d’écran 2024-06-20 à 01 16 44" src="https://github.com/immich-app/immich/assets/30565737/63c8e3ad-8975-4e7f-81d9-623745892718">

<img width="646" alt="image" src="https://github.com/immich-app/immich/assets/30565737/d72e7dc2-fec5-4741-9e6a-6d0a82be22a0">

so i'm wondering, if the storage template is good, my upload folders and paths are correct, the file is named correctly everywhere in the folder structure, which file is downloaded ? can't find anywhere the IMG files

<img width="646" alt="Capture d’écran 2024-06-20 à 01 33 57" src="https://github.com/immich-app/immich/assets/30565737/83c6cfbf-0d04-4542-8da9-20833fcb81d9">

thanks

**Comment by bo0tzz** [comment]
When you upload a file, the name it has at that time is stored in the database as the original filename. If you download the file from the webui that's the name it gets given.

**Reply by yarez0** [reply]
thanks, so immich rename the file on the fly when the download is asked wright ?

**Reply by bo0tzz** [reply]
Yes

**Reply by yarez0** [reply]
know why ? i wanted to use the storage template to automatically rename all my files but if the download option give them another name it's useless

**Reply by yarez0** [reply]
have to upload / auto rename / delete from album / re upload with the storage template name from the library path
